src/Playwright/Transport/Connection.cs

```diff
@@ -305,8 +305,7 @@ internal void Dispatch(PlaywrightServerMessage message)
                 @object.DisposeOwner(message.Params.Value.TryGetProperty("reason", out var reason) ? reason.GetString() : null);
                 return;
             }
-
-            @object.Channel?.OnMessage(message.Method, message.Params);
+            ((ChannelOwnerBase)@object).OnMessage(message.Method, message.Params);
```

Comment:
Maybe we can change Objects.TryGetValue to return ChannelOwnerBase and avoid this cast?